File: src/bindings/csharp/test/sbml/TestConstraint.cs
/// 
///  @file    TestConstraint.cs
///  @brief   SBML Constraint unit tests
///  @author  Frank Bergmann (Csharp conversion)
///  @author  Akiya Jouraku (Csharp conversion)
///  @author  Sarah Keating 
/// 
///  $Id$
///  $HeadURL$
/// 
///  This test file was converted from src/sbml/test/TestConstraint.c
///  with the help of conversion sciprt (ctest_converter.pl).
/// 
/// <!---------------------------------------------------------------------------
///  This file is part of libSBML.  Please visit http://sbml.org for more
///  information about SBML, and the latest version of libSBML.
/// 
///  Copyright 2005-2010 California Institute of Technology.
///  Copyright 2002-2005 California Institute of Technology and
///                      Japan Science and Technology Corporation.
///  
///  This library is free software; you can redistribute it and/or modify it
///  under the terms of the GNU Lesser General Public License as published by
///  the Free Software Foundation.  A copy of the license agreement is provided
///  in the file named "LICENSE.txt" included with this software distribution
///  and also available online as http://sbml.org/software/libsbml/license.html
/// --------------------------------------------------------------------------->*/


namespace LibSBMLCSTest {

  using libsbml;

  using  System.IO;

  public class TestConstraint {
    public class AssertionError : System.Exception 
    {
      public AssertionError() : base()
      {
        
      }
    }


    static void assertTrue(bool condition)
    {
      if (condition == true)
      {
        return;
      }
      throw new AssertionError();
    }

    static void assertEquals(object a, object b)
    {
      if ( (a == null) && (b == null) )
      {
        return;
      }
      else if ( (a == null) || (b == null) )
      {
        throw new AssertionError();
      }
      else if (a.Equals(b))
      {
        return;
      }
  
      throw new AssertionError();
    }

    static void assertNotEquals(object a, object b)
    {
      if ( (a == null) && (b == null) )
      {
        throw new AssertionError();
      }
      else if ( (a == null) || (b == null) )
      {
        return;
      }
      else if (a.Equals(b))
      {
        throw new AssertionError();
      }
    }

    static void assertEquals(bool a, bool b)
    {
      if ( a == b )
      {
        return;
      }
      throw new AssertionError();
    }

    static void assertNotEquals(bool a, bool b)
    {
      if ( a != b )
      {
        return;
      }
      throw new AssertionError();
    }

    static void assertEquals(int a, int b)
    {
      if ( a == b )
      {
        return;
      }
      throw new AssertionError();
    }

    static void assertNotEquals(int a, int b)
    {
      if ( a != b )
      {
        return;
      }
      throw new AssertionError();
    }

    private Constraint C;

    public void setUp()
    {
      C = new  Constraint(2,4);
      if (C == null);
      {
      }
    }

    public void tearDown()
    {
      C = null;
    }

    public void test_Constraint_create()
    {
      assertTrue( C.getTypeCode() == libsbml.SBML_CONSTRAINT );
      assertTrue( C.getMetaId() == "" );
      assertTrue( C.getNotes() == null );
      assertTrue( C.getAnnotation() == null );
      assertEquals( false, C.isSetMessage() );
      assertEquals( false, C.isSetMath() );
    }

    public void test_Constraint_createWithNS()
    {
      XMLNamespaces xmlns = new  XMLNamespaces();
      xmlns.add( "http://www.sbml.org", "testsbml");
      SBMLNamespaces sbmlns = new  SBMLNamespaces(2,2);
      sbmlns.addNamespaces(xmlns);
      Constraint object1 = new  Constraint(sbmlns);
      assertTrue( object1.getTypeCode() == libsbml.SBML_CONSTRAINT );
      assertTrue( object1.getMetaId() == "" );
      assertTrue( object1.getNotes() == null );
      assertTrue( object1.getAnnotation() == null );
      assertTrue( object1.getLevel() == 2 );
      assertTrue( object1.getVersion() == 2 );
      assertTrue( object1.getNamespaces() != null );
      assertTrue( object1.getNamespaces().getLength() == 2 );
      object1 = null;
    }

    public void test_Constraint_free_NULL()
    {
    }

    public void test_Constraint_setMath()
    {
      ASTNode math = libsbml.parseFormula("2 * k");
      C.setMath(math);
      assertTrue( C.getMath() != math );
      assertEquals( true, C.isSetMath() );
      C.setMath(C.getMath());
      assertTrue( C.getMath() != math );
      C.setMath(null);
      assertEquals( false, C.isSetMath() );
      if (C.getMath() != null);
      {
      }
      math = null;
    }

    public void test_Constraint_setMessage()
    {
      XMLNode text = XMLNode.convertStringToXMLNode(" Some text ",null);
      XMLTriple triple = new  XMLTriple("p", "http://www.w3.org/1999/xhtml", "");
      XMLAttributes att = new  XMLAttributes();
      XMLNamespaces xmlns = new  XMLNamespaces();
      xmlns.add( "http://www.w3.org/1999/xhtml", "");
      XMLNode p = new XMLNode(triple,att,xmlns);
      p.addChild(text);
      XMLTriple triple1 = new  XMLTriple("message", "", "");
      XMLAttributes att1 = new  XMLAttributes();
      XMLNode node = new XMLNode(triple1,att1);
      node.addChild(p);
      C.setMessage(node);
      assertTrue( C.getMessage() != node );
      assertTrue( C.isSetMessage() == true );
      C.setMessage(C.getMessage());
      assertTrue( C.getMessage() != node );
      assertTrue( C.getMessageString() != null );
      C.unsetMessage();
      assertEquals( false, C.isSetMessage() );
      if (C.getMessage() != null);
      {
      }
      node = null;
    }

  }
}

File: src/bindings/ruby/test/sbml/TestConstraint.rb
#
# @file    TestConstraint.rb
# @brief   SBML Constraint unit tests
#
# @author  Akiya Jouraku (Ruby conversion)
# @author  Sarah Keating 
#
# $Id$
# $HeadURL$
#
# This test file was converted from src/sbml/test/TestConstraint.c
# with the help of conversion sciprt (ctest_converter.pl).
#
#<!---------------------------------------------------------------------------
# This file is part of libSBML.  Please visit http://sbml.org for more
# information about SBML, and the latest version of libSBML.
#
# Copyright 2005-2009 California Institute of Technology.
# Copyright 2002-2005 California Institute of Technology and
#                     Japan Science and Technology Corporation.
# 
# This library is free software; you can redistribute it and/or modify it
# under the terms of the GNU Lesser General Public License as published by
# the Free Software Foundation.  A copy of the license agreement is provided
# in the file named "LICENSE.txt" included with this software distribution
# and also available online as http://sbml.org/software/libsbml/license.html
#--------------------------------------------------------------------------->*/
require 'test/unit'
require 'libSBML'

class TestConstraint < Test::Unit::TestCase

  def setup
    @@c = LibSBML::Constraint.new()
    if (@@c == nil)
    end
  end

  def teardown
    @@c = nil
  end

  def test_Constraint_create
    assert( @@c.getTypeCode() == LibSBML::SBML_CONSTRAINT )
    assert( @@c.getMetaId() == "" )
    assert( @@c.getNotes() == nil )
    assert( @@c.getAnnotation() == nil )
    assert_equal false, @@c.isSetMessage()
    assert_equal false, @@c.isSetMath()
  end

  def test_Constraint_createWithLevelVersionAndNamespace
    xmlns = LibSBML::XMLNamespaces.new()
    xmlns.add( "http://www.sbml.org", "sbml")
    object = LibSBML::Constraint.new(2,1,xmlns)
    assert( object.getTypeCode() == LibSBML::SBML_CONSTRAINT )
    assert( object.getMetaId() == "" )
    assert( object.getNotes() == nil )
    assert( object.getAnnotation() == nil )
    assert( object.getLevel() == 2 )
    assert( object.getVersion() == 1 )
    assert( object.getNamespaces() != "" )
    assert( object.getNamespaces().getLength() == 1 )
    object = nil
  end

  def test_Constraint_createWithMath
    math = LibSBML::parseFormula("1 + 1")
    c = LibSBML::Constraint.new(math)
    assert( c.getTypeCode() == LibSBML::SBML_CONSTRAINT )
    assert( c.getMetaId() == "" )
    assert( c.getMath() != math )
    assert_equal false, c.isSetMessage()
    assert_equal true, c.isSetMath()
    c = nil
  end

  def test_Constraint_free_NULL
  end

  def test_Constraint_setMath
    math = LibSBML::parseFormula("2 * k")
    @@c.setMath(math)
    assert( @@c.getMath() != math )
    assert_equal true, @@c.isSetMath()
    @@c.setMath(@@c.getMath())
    assert( @@c.getMath() != math )
    @@c.setMath(nil)
    assert_equal false, @@c.isSetMath()
    if (@@c.getMath() != nil)
    end
    math = nil
  end

  def test_Constraint_setMessage
    node = LibSBML::XMLNode.new()
    @@c.setMessage(node)
    assert( @@c.getMessage() != node )
    assert( @@c.isSetMessage() == true )
    @@c.setMessage(@@c.getMessage())
    assert( @@c.getMessage() != node )
    assert( @@c.getMessageString() != nil )
    assert( ( "" != @@c.getMessageString() ) == false )
    @@c.unsetMessage()
    assert_equal false, @@c.isSetMessage()
    if (@@c.getMessage() != nil)
    end
    node = nil
  end

end
